### ISSUE TITLE
Menu: reconsider the modal behavior

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#68830]
### Description

Splitting this out from https://github.com/WordPress/gutenberg/issues/67078, which reports a few more issues with the Menu component.

The Menu component implementation is based on the Ariakit library and, so far, it uses a modal behavior. That is, when a meny is open, interaction with the rest of the page is prevented. There have already been discussions on whether to use the modal behavior for the Menu but they occurred in more general issues while I would like to have a discussion specifically on this point.

The more I test this behavior, the more I'm confused by it.

For example, in the Site editor > Templates page, I was confused and staring at the screen because I couldn't understand why spme buttons didn't do anything when clicking them. I had to click twice to make them do something.

Actually, like for a modal dialog, the Menu modal behavior does add an overlay 'backdrop' to prevent intereaction with the rest of the page.

But...

the overlay is transparent. Users don't have any visual indication that the interaction with the rest of the page is _intentionally_ prevented. As such, clicking on other controls in the UI and seeing that nothing happens is very confusing. This is very different from, for example, the odal behavior of a dialog where the overlay backdrop is _visible_.

Video to better illustrate.
At some point, I'm adding some background color to the overlay to better illustrate when it prevents interacting with the rest of the page.

https://github.com/user-attachments/assets/2881997d-e7f4-4c12-abd9-687ef885939a

Overall, I think the user experience is very confusing.
I'd like to reconsider whether the modal behavior adds any benefit for users, including keyboard users and assistive technology users.

### Step-by-step reproduction instructions

- Go to Site editor > Templates.
- Click the ellipsis button of a template card.
- A dropdown menu opens. Keep it open.
- Click the ellipsis button of _another_ template card.
- Observe that nothing happens.
- Click again.
- Observe this time something happens (the othe rmenu opens).
- Click the preview of another template card.
- Observe that nothing happens.
- Click again.
- Observe this time something happens.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by Rishit30G** [comment]
Thanks for sharing the issue @afercia, 

I'm sharing a screencast which introduce a new modal backdrop, to me it looks like a proper modal which gives the user an indication that the other options on the page are not selectable 
Would appreciate your thoughts on this one 🙇🏻 

Thanks 

https://github.com/user-attachments/assets/9ae8a126-0873-477b-acbc-deb64a73db8d

**Comment by afercia** [comment]
Thanks @Rishit30G for your feedback and example. Personally, I think the modal behavior should be either entirely removed or kept but in this case it should provide a visual indication e.g. the overlay should be visible like in your example. However, I'm not sure why a menu should prevent interaction with the rest of the page in the first place.

I guess the current implementation is inspired by the interaction with menus in some operating systems. For example, on macOS, when a menu of the operating system is open then interacting with the currently active window is prevented. That makes sense because the UI of the menu in the operating system and the active window are clearly separated. Video:

https://github.com/user-attachments/assets/9f1fc35d-35a8-4e54-be25-00d2d1cd25f8

Things become less consistent when the active window itself has an open menu and trying to interact with a control in the rest of the active window. Quickly testing with browsers:
- Chrome: open a Chrome menu and try to click a link in the rest of the page: interaction is prevented (the menu is modal).
- Firefox: open a Firefox menu and try to click a link in the rest of the page: the link is clickable (the menu isn't modal).
- Safari: same as Firefox.

That's to say that if this implementation has been inspired by some existing behaviors in operating systems, that source of inspiration is inconsistent in the first place and I'm not even sure other operating systems use the same pattern.

Regardless, I'd tend to think the users expectation would be the interaction with the rest of the page to _not_ be prevented.

Worth also noting the ARIA [menu bar](https://www.w3.org/WAI/ARIA/apg/patterns/menubar/) and [menu button](https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/) examples don't use any modal behavior.

Before any coding, I'd think the reasons why the menu current implementation is modal should be investigated and see if the arguments for the modality are still valid.

**Comment by Rishit30G** [comment]
Points well stated, thanks for clear explanation and examples on the modal behaviour! 🙇🏻 

Edit: I tried removing the overlay, just to check the functionality but seems like it's not removing, or there is some other way to do it 🤔 

https://github.com/user-attachments/assets/64640bbd-aff4-4be7-b651-543b39434a94

**Comment by afercia** [comment]
@Rishit30G have you tried to set the Menu [modal prop](https://github.com/WordPress/gutenberg/blob/a411f333ae717961749714064b35c373f2f0b6c1/packages/components/src/menu/types.ts#L73) to false?

The Menu implementation is based on the external library Ariakit, so the modal behavior may be an Ariakit internal implementation.

**Comment by Rishit30G** [comment]
Thanks @afercia! 🙇🏻 
The implementation shared works, now we have the access to the other elements even when the menu is open 
Now based on the final discussion we should come to a conclusion that either we don't need the modal or we need a modal with a light dark opaque backdrop 

https://github.com/user-attachments/assets/3e08d82b-eeae-4431-b285-7226f5f1d5d5

**Comment by afercia** [comment]
Cc @joedolson can this be discussed at the next accessibility bug-scrub?

**Comment by joedolson** [comment]
We discussed this in the accessibility bug scrub today, and agreed that these modals should not be modal. The modal behavior is unnecessary, and while adding an opaque overlay would clarify the behavior, it would be better if the menus did not behave as modals.

**Comment by Rishit30G** [comment]
I have raised a PR to fix the modal behaviour @afercia, thanks! 🙇🏻

**Comment by carolinan** [comment]
It may be good to clarify which menus. The `<Menu.Popover>` is also used outside the Data views.

**Comment by mirka** [comment]
> Before any coding, I'd think the reasons why the menu current implementation is modal should be investigated and see if the arguments for the modality are still valid.

It was a close call, but I remember the biggest reason we went with `modal` here is to avoid having to deal with visual issues when the page is scrolled with the popover open. There are a lot of complications that can happen when scrolling is involved, especially with our current z-index-based popover layering system (#71463). Enabling `modal` disables scrolling altogether, so we are able to avoid that class of problem entirely.

You can see a potential complication even in this GitHub issue page, where dropdowns in the right sidebar are scrolling together with the main content container, when the sidebar itself is sticky. You can also imagine dropdowns scrolling underneath some kind of sticky header, rather than above it. Not saying these are unsolvable problems, but they are a class of problem we can bypass if we just stopped the page from scrolling.

Given that a lot of dropdowns do behave modally (macOS menus, `select`s), we were hoping it would acceptable.
